lib/browser/api/desktop-capturer.ts

```diff
@@ -62,20 +64,45 @@ export async function getSources (args: Electron.SourcesOptions) {
               win.resizable = resizableValues.get(win.id);
             }
           };
-        }
+          // On Windows, the underlying WebRTC implementation does not return sources
+          // originating owned by the current process due to a Windows deadlock issue.
+          // CL: https://chromium-review.googlesource.com/c/chromium/src/+/2907415
+        } else if (process.platform === 'win32' && captureWindow) {
+          const fetches = BrowserWindow.getAllWindows().map(async (win) => {
+            let thumbnail = null;
+            if (thumbnailSize.width > 0 && thumbnailSize.height > 0) {
+              const pageContents = await win.capturePage();
+              thumbnail = pageContents.resize(thumbnailSize);
+            } else {
+              thumbnail = nativeImage.createEmpty();
+            }
+            return {
+              name: win.getTitle(),
+              id: win.getMediaSourceId(),
+              thumbnail: thumbnail,
+              display_id: '',
+              appIcon: null
+            };
+          });
+          winsOwnedByElectronProcess = await Promise.all(fetches);
+        };
       }
       // Remove from currentlyRunning once we resolve or reject
       currentlyRunning = currentlyRunning.filter(running => running.options !== options);
     };
 
-    capturer._onerror = (error: string) => {
-      stopRunning();
+    capturer._onerror = async (error: string) => {
```

Comment:
This will leave a dangling promise, this shouldn't be `async`.

Instead `winsOwnedByElectronProcess` should be `Promise<Array<>>` instead of `Array`